lightningd/opening_control.c

```diff
@@ -371,6 +371,12 @@ static void opening_funder_finished(struct subd *openingd, const u8 *resp,
 	/* Extract the change output and add it to the DB */
 	wallet_extract_owned_outputs(ld->wallet, fundingtx, NULL, &change_satoshi);
 
+	/* Make sure we recognize our change output by its scriptpubkey in
+	 * future. This assumes that we have only two outputs, may not be true
+	 * if we add support for multifundchannel */
+	if (tal_count(fundingtx->output) == 2)
+		txfilter_add_scriptpubkey(ld->owned_txfilter, fundingtx->output[!funding_outnum].script);
+
```

Comment:
I'm a bit confused... since this spend is to our own wallet, don't we watch it anyway, just as if someone else send to that address?

Reply:
Shouldn't we be watching all known keyindexes all the time? Incrementing the key index should also mean watching for any outputs. Seems simplest and safest?

Reply:
Indeed we should, but is it better to push the scriptpubkey generation and registration in the filter down into the keyindex generation? Alternatively we could add a new function that takes care of that. It may need to tear `funding_tx` open to inject the scriptpubkey generated externally.